src/js/utils/linked-list.js

```diff
@@ -0,0 +1,84 @@
+export default class LinkedList {
+  constructor(options) {
+    this.head = null;
+    this.tail = null;
+    if (options) {
+      let {adoptItem, freeItem} = options;
+      this.adoptItem = adoptItem;
+      this.freeItem = freeItem;
+    }
+  }
+  prepend(item) {
+    this.insertBefore(item, this.head);
+  }
+  append(item) {
+    this.insertBefore(item, null);
+  }
+  insertAfter(item, prevItem) {
+    let nextItem = null;
+    if (prevItem) {
+      nextItem = prevItem.next;
+    }
+    this.insertBefore(item, nextItem);
+  }
+  insertBefore(item, nextItem) {
```

Comment:
Actually that ^^ is still wrong. I had to pull this and branch off it to make sure I had it right. Here's https://github.com/mixonic/content-kit-editor/pull/1, which merges onto this PR, if you like where those ^^ comments were going.